core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -125,6 +125,10 @@ private TableIdentifier canonicalizeIdentifier(TableIdentifier tableIdentifier)
     }
   }
 
+  public Catalog catalog() {
```

Comment:
Catalog available for a call procedure can be a Caching catalog. So, need to expose the underlying catalog to check whether it is Hadoop catalog or not to block the feature for hadoop catalog.